### PR TITLE
Polish new snippet form

### PR DESCRIPTION
**Pull request by alex-bu-93**

### CODE REVIEW

**Comment by alex-bu-93** [comment]
Please leave any comments! I will be glad to fix everything :)

File: angular.json [hunk 1]
```diff
@@ -32,9 +32,17 @@
...
             "styles": [
-              "apps/codelab/src/styles.scss"
+              "apps/codelab/src/styles.scss",
+              "node_modules/prismjs/themes/prism-okaidia.css",
+              "node_modules/prismjs/plugins/line-numbers/prism-line-numbers.css"
             ],
-            "scripts": []
+            "scripts": [
+              "node_modules/marked/lib/marked.js",
+              "node_modules/prismjs/prism.js",
```
Comment (kirjs) [inline angular.json]: Would this include all this stuff for the codelab page?
We may need a separate app

File: apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts [hunk 2]
```diff
@@ -1,50 +1,164 @@
...
+import { MARKDOWN_PLACEHOLDER, TAGS_LIST, angularSampleCode } from '../shared';
+import { SnippetOverviewComponent } from './snippet-modal/snippet-overview.component';
+import { Observable } from 'rxjs/internal/Observable';
+import { map, startWith } from 'rxjs/operators';
 
 @Component({
   selector: 'codelab-create-snippet',
   templateUrl: './create-snippet.component.html',
-  styleUrls: ['./create-snippet.component.css']
+  styleUrls: ['./create-snippet.component.scss']
```
Comment (kirjs) [inline apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts]: OnPush

File: apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts [hunk 3]
```diff
@@ -1,50 +1,164 @@
...
-    content: ['', Validators.required],
-    author: [''],
-    level: ['beginner'],
-    tags: [''],
-    bonus: [''],
-    links: [''],
+    level: ['Beginner', Validators.required],
+    tags: ['Tips', Validators.required],
+    content: [MARKDOWN_PLACEHOLDER, Validators.required],
+    bonus: [MARKDOWN_PLACEHOLDER, ''],
```
Comment (kirjs) [inline apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts]: Let's have it empty?

File: apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts [hunk 4]
```diff
@@ -1,50 +1,164 @@
...
+    links: [MARKDOWN_PLACEHOLDER, ''],
```
Comment (kirjs) [inline apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts]: Let's have a link example

File: apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts [hunk 5]
```diff
@@ -1,50 +1,164 @@
...
-      this.snippet = {
-        ...raw,
-        bonus: raw.bonus.trim(),
-        links: raw.links.trim(),
-        tags: raw.tags.split(',')
-      };
-
-      if (raw.demo['app.component.ts'] !== this.initialCode['app.component.ts']) {
-        this.snippet.componentCode = raw.demo['app.component.ts'];
+  isBonus = false;
```
Comment (kirjs) [inline apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts]: eng: hasBonus
Same for below

File: apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts [hunk 6]
```diff
@@ -1,50 +1,164 @@
...
+  isLinks = false;
+  isDemo = false;
+
+  separatorKeysCodes: number[] = [ENTER, COMMA];
+
+  constructor(
+    private fb: FormBuilder,
+    public dialog: MatDialog
+  ) {
+    this.filteredTags = this.snippetForm.get('tags').valueChanges.pipe(
```
Comment (kirjs) [inline apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts]: Can this be moved out of component?

File: apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts [hunk 7]
```diff
@@ -1,50 +1,164 @@
...
+    value['bonus'] = this.isBonus ? value['bonus'] : null;
+    value['links'] = this.isLinks ? value['links'] : null;
+    value['demo'] = this.isDemo ? value['demo'] : null;
+    const isDemoComponentChanged = this.isDemo && value.demo['app.component.ts'] !== angularSampleCode['app.component.ts'];
+    const isDemoModuleChanged = this.isDemo && value.demo['app.module.ts'] !== angularSampleCode['app.module.ts'];
+
+    return `
+---
+
+## __Title:__ ${value.title}
```
Comment (kirjs) [inline apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts]: pls update this to 
---
title:  xx
level: xx
...
---

File: apps/codelab/src/app/codelabs/angular-thirty-seconds/shared/constatns.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
+export const MARKDOWN_PLACEHOLDER = `## Markdown __rulez__!
```
Comment (kirjs) [inline apps/codelab/src/app/codelabs/angular-thirty-seconds/shared/constatns.ts]: We actually shouldn't allow any headers on the content

File: apps/codelab/src/app/codelabs/angular-thirty-seconds/shared/constatns.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+\`\`\`typescript
+
+### Syntax highlight
```
Comment (kirjs) [inline apps/codelab/src/app/codelabs/angular-thirty-seconds/shared/constatns.ts]: Let's have something that's more like a real content example here

File: apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts (hunk 6 above)
Comment (alex-bu-93) [reply apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts]: It was taken exactly from here: https://material.angular.io/components/chips/examples

Please, explain where do you want to see it? :)

Comment (NothingEverHappens) [reply apps/codelab/src/app/codelabs/angular-thirty-seconds/create-snippet/create-snippet.component.ts]: sorry, I mean constructor

**Pull request merged**